compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2810,6 +2848,24 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 rustc_errors::struct_span_err!(tcx.sess, self_ty.span, E0782, "{}", msg)
                     .multipart_suggestion_verbose(label, sugg, Applicability::MachineApplicable)
                     .emit();
+                self.maybe_lint_blanket_trait_impl(
+                    &self_ty,
+                    |fn_tcx, ty, blanket_msg, help_msg, blanket_sugg| {
+                        rustc_errors::struct_span_err!(
```

Comment:
I don't know if this should be an additional `struct_span_err!` -- it should just be adding onto the previous one, I think.

Reply:
Same with below.

Reply:
Mh the span_error should be easy and I think is a good improvement! however, there is any way to do this with the lint?

There is the following problem

```
error: unused variable: `help_msg`
    --> compiler/rustc_typeck/src/astconv/mod.rs:2809:21
     |
2809 |                 let help_msg = "use blanket implementation";
     |                     ^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_help_msg`

error[E0382]: use of moved value: `lint`
    --> compiler/rustc_typeck/src/astconv/mod.rs:2871:81
     |
2863 |                     |lint| {
     |                      ---- move occurs because `lint` has type `LintDiagnosticBuilder<'_, ()>`, which does not implement the `Copy` trait
2864 |                         lint.build(msg)
     |                              ---------- `lint` moved due to this method call
...
2871 |                         self.maybe_lint_blanket_trait_impl::<()>(&self_ty, &mut lint.build(msg));
     |                                                                                 ^^^^ value used here after move

```

Reply:
You need to call `lint.build` once.

```rust
let mut diag = lint.build(msg);
self.maybe_lint_blanket_trait_impl(&self_ty, &mut diag);
// do whatever the previous `lint.build().a().b().c().emit();` using `diag`.
```

Reply:
Ok so in this case, we have to need to come up with a better help message in the case the here? a generic one otherwise we need to change the code to have the confirmation to be in a blanket case before build the lint.

Reply:
during the coding I found your logic, thanks for your patience